lib/spack/spack/compilers/nag.py

```diff
@@ -20,6 +21,11 @@ class Nag(Compiler):
                    'f77' : 'nag/nagfor',
                    'fc'  : 'nag/nagfor' }
 
+    @property
+    def openmp_flag(self):
+        return "-openmp"
+
+
```

Comment:
This is the last change, I swear! Until better mixed compiler support is implemented, can you add:
```python
@property
def cxx11_flag(self):
    # NAG does not have a C++ compiler
    # However, it can be mixed with a compiler that does support it
    return "-std=c++11"
```
With the current implementation in compilers.py, it will raise an error message, even if I try to combine NAG and GCC. Honestly, this is kind of a bug, and we should work on better mixed compiler support. Ideally, we would check the package of the C++ compiler to determine the appropriate cxx_flag.

@tgamblin: This is yet another reason why compilers should be a build dependency. A package could theoretically `depends_on('cc'); depends_on('cxx')` and then each compiler would have its own package specifying what it provides and what flags to use. By specifying exactly what compilers a package depends on, we could know when to use `FC` instead of both `FC` and `F77` (some packages complain about this). We could also catch bugs where someone tries to use a compiler without Fortran, but the package requires a Fortran compiler. Imagine using `depends_on('fortran', when='+fortran')` for the HDF5 package.